waspc/src/Wasp/Lib.hs

```diff
@@ -37,64 +39,74 @@ compile ::
   CompileOptions ->
   IO ([CompileWarning], [CompileError])
 compile waspDir outDir options = do
-  (analyzerWarnings, appSpecOrAnalyzerErrors) <- analyzeWaspProject waspDir options
+  appSpecOrAnalyzerErrors <- analyzeWaspProject waspDir options
+  analyzerWarnings <- warnIfDotEnvPresent waspDir
```

Comment:
Why did you take this out of analyzeWaspProject?

Reply:
Ah, I guess why you moved it now, it was messing up your ExceptT approach.
But I am not sure if that is the right reason to move it. We could still put it back, I am sure, while going with ExceptT nevertheless.

Reply:
[This comment](https://github.com/wasp-lang/wasp/pull/753#discussion_r997582609) kinda covers my reasoning. But yes, it was also in part to make everything nicely fit with my transformer :see_no_evil:.

Still, even without the `ExceptT`, I'd still leave it here for the reasons in the comment. Btw, I came up with a hopefully better name for `analyzeWaspProject` that justifies it.

Reply:
I changed the name of the function, hopefully now it makes more sense.